tests/MongoDB.Driver.Tests/Specifications/client-side-encryption/ClientEncryptionProseTests.cs

```diff
@@ -0,0 +1,827 @@
+            bool withExternalKeyVault = false,
+            Action<ClusterBuilder> clusterConfigurator = null)
+        {
+            var extraOptions = new Dictionary<string, object>()
+            {
+                { "mongocryptdSpawnPath", Environment.GetEnvironmentVariable("MONGODB_BINARIES") ?? string.Empty } 
+            };
+
+            var mongoClientSettings = new MongoClientSettings
+            {
+                GuidRepresentation = GuidRepresentation.Unspecified,
+                ClusterConfigurator = clusterConfigurator
+            };
+
+            if (keyVaultNamespace != null || schemaMapDocument != null || kmsProviders != null || withExternalKeyVault)
+            {
+                Dictionary<string, BsonDocument> schemaMap = null;
+                if (schemaMapDocument != null)
+                {
+                    var element = schemaMapDocument.Single();
+                    schemaMap = new Dictionary<string, BsonDocument>
+                    {
+                        { element.Name, element.Value.AsBsonDocument }
+                    };
+                }
+
+                if (kmsProviders == null)
+                {
+                    kmsProviders = new ReadOnlyDictionary<string, IReadOnlyDictionary<string, object>>(new Dictionary<string, IReadOnlyDictionary<string, object>>());
+                }
+
+                var autoEncryptionOptions = new AutoEncryptionOptions(
+                    keyVaultNamespace,
+                    kmsProviders,
+                    schemaMap: Optional.Create<IReadOnlyDictionary<string, BsonDocument>>(schemaMap),
+                    extraOptions: extraOptions);
+
+                if (withExternalKeyVault)
+                {
+                    var externalKeyVaultClient = 
+                        new MongoClient(
+                            new MongoClientSettings
+                            {
+                                Credential = MongoCredential.FromComponents(null, null, "fake-user", "fake-pwd")
+                            });
+                    autoEncryptionOptions = autoEncryptionOptions
+                        .With(
+                            keyVaultClient: Optional.Create<IMongoClient>(externalKeyVaultClient));
+                }
+
+                mongoClientSettings.AutoEncryptionOptions = autoEncryptionOptions;
+            }
+
+            return new MongoClient(mongoClientSettings);
+        }
+
+        private void Insert(
+            IMongoCollection<BsonDocument> collection,
+            bool async,
+            params BsonDocument[] documents)
+        {
+            if (async)
+            {
+                collection
+                    .InsertManyAsync(documents)
+                    .GetAwaiter()
+                    .GetResult();
+            }
+            else
+            {
+                collection.InsertMany(documents);
+            }
+        }
+
+        public class JsonTestDataFactory : EmbeddedResourceJsonFileReader
```

Comment:
This is not a test data factory (at least not in the Xunit sense).

Let's name this something more accurate like "JsonFileReader".

The name probably doesn't have to be longer than that because it is a nested class, so the full name is actually `...ClientEncryptionProseTests.JsonFileReader`.